DOCS-2878: Revise payment component documentation (#2441)
MultiSafepay/docs: content/checkout/single.md

    <br>

    | Method | Description |
    | ---- | ---- |
    |`getErrors`| Returns error details, e.g. error messages or codes|
    |`hasErrors`| Returns a boolean value depending on whether errors have been registered |

>     |`getPaymentData`| Returns a `payment_data` object with a `payload` containing the customer's payment details, used to [create orders](/docs/payment-component-single#3-create-an-order), and the `gateway`.|

Reviewer: Is the `gateway` returned as part of the `payment_data` object?

Author: Yes

Reviewer: Then how about:

```suggestion
    |`getPaymentData`| Returns a `payment_data` object containing the `gateway`, and a `payload` containing the customer's payment details, used to [create orders](/docs/payment-component-single#3-create-an-order).
```

    </details>

# 3. Create an order

## Handle the interaction

1. Assign the button element to a variable:

    ``` js
    const paymentButton = document.querySelector('#payment-button');
    ```

2. Create an event handler for the payment button to:

    - When the customer clicks the payment button, call the component's `getPaymentData()` method.
    - Send the response to your server and [create an order](#create-an-order).
    - Return the reponse from your server to the client-side to redirect the customer.
    <br>

### Redirect the customer

The component's `redirection` handler redirects the customer to the relevant page:

- If the customer actions are required to complete the payment (e.g. by completing 3D Secure or iDEAL issuer authentication), the customer is redirected to the relevant page. If successful, the customer is then redirected to the `redirect_url` (i.e. the 'success page'). 
- If no customer actions are required to complete the payment, the customer is then redirected to the `redirect_url` (i.e. the 'success page').
- If the customer chooses to pay by bank transfer, the component displays the banking details for customers to complete the payment. 

> - If a QR code is available for customers to complete the payment on their mobile device, the component displays the QR code. 

Reviewer: ```suggestion
- If a QR code is available for customers to complete payment on their mobile device, the component displays the QR code. 
```

  
### Avoid duplicate orders

If the customer clicks your payment button during the latency before redirection, this creates duplicate orders. 

> To avoid duplicate orders, disable the button until you have attempted to create an order.  
> Then, check `response.success`:

Reviewer: ```suggestion
To avoid duplicate orders, disable the button until you have attempted to create an order. 

Then, check `response.success`:
```

> - If `true`, don't re-enable the button and proceed to the redirect.

Reviewer: ```suggestion
- If `true`, don't re-enable the button, and proceed to the redirect.
```

2. Check that `response.success` is `true`.

3. Handle the response

    Call the `PaymentComponent.init()` method using the following arguments:

    ``` js
    PaymentComponent.init('redirection', {
        order: response.data
    });
    ```

    The component's `redirection` handler redirects the customer to the relevant page:
    

>     - If the customer actions are required to complete the payment (e.g. by completing 3D Secure or iDEAL issuer authentication), the customer is redirected to the relevant page. If successful, the customer is then redirected to the `redirect_url` (i.e. the 'success page'). 

Author: ```suggestion
    - If customer action is required to complete payment (e.g. by completing 3D Secure or iDEAL issuer authentication), the customer is redirected to the relevant page. If successful, the customer is then redirected to the `redirect_url` (i.e. the "success page"). 
```

>     - If no customer actions are required to complete the payment, the customer is then redirected to the `redirect_url` (i.e. the 'success page').

Author: ```suggestion
    - If no customer action is required to complete payment, the customer is redirected to the `redirect_url` (i.e. the "success page").
```

>     - If the customer chooses to pay by bank transfer, the component displays the banking details for customers to complete the payment. 

Author: ```suggestion
    - If the customer chooses to pay by bank transfer, the component displays the banking details for customers to complete payment. 
```

>     - If a QR code is available for customers to complete the payment on their mobile device, the component displays the QR code. 

Reviewer: See above comments on the same information.

>     - If a QR code is available for customers to complete the payment on their mobile device, the component displays the QR code. 

Author: ```suggestion
    - If a QR code is available for customers to complete payment on their mobile device, the component displays the QR code. 
```

>     - If a QR code is available for customers to complete the payment on their mobile device, the component displays the QR code. 

Author: ```suggestion
    - If a QR code is available for customers to complete payment on their mobile device, the component displays the QR code. 
```

# 4. Go live

To test the payment method, use our [Testing](/docs/testing#test-payment-details).

When you're ready to process real payments, make the following changes:

> 1. In [Step 1: Add the elements](#1-add-the-elements), replace test JavaScript library with the live JavaScript library:

Reviewer: ```suggestion
1. In [Step 1: Add the elements](#1-add-the-elements), replace the test JavaScript library with the live JavaScript library:
```
